### PR TITLE
codes push for curve adapter

### PR DESCRIPTION
**Pull request by naruto0913**
PR for polygon curve adapter

### CODE REVIEW

**Review by daiki-iu — CHANGES_REQUESTED** [review]
@naruto0913  LGTM!
I need some updates in this PR.
- build a separate HedgepieLibraryMatic
- build a separate AdapterManger

File: v1-contract/contracts/HedgepieAdapterManager.sol [hunk 1]
```diff
@@ -124,6 +124,36 @@ contract HedgepieAdapterManager is Ownable {
...
+        returns (
+            address to,
+            uint256 value,
+            bytes memory data
+        )
+    {
+        require(_amount > 0, "Amount can not be 0");
+        return IAdapter(_adapter).getRemoveLiqCallData(_amount);
+    }
+
```
Comment (daiki-iu) [inline v1-contract/contracts/HedgepieAdapterManager.sol:156]: @naruto0913  we don't need these functions here.

File: v1-contract/contracts/HedgepieInvestor.sol [hunk 2]
```diff
@@ -1,14 +1,9 @@
...
 
-import "./interfaces/IYBNFT.sol";
-import "./interfaces/IAdapter.sol";
-import "./interfaces/IVaultStrategy.sol";
-import "./interfaces/IAdapterManager.sol";
-import "./interfaces/IPancakePair.sol";
-import "./interfaces/IPancakeRouter.sol";
+import "./libraries/HedgepieLibrary.sol";
 
 contract HedgepieInvestor is Ownable, ReentrancyGuard {
```
Comment (daiki-iu) [inline v1-contract/contracts/HedgepieInvestor.sol:8]: @naruto0913  we don't need any change in HedgepieInvestor contract

File: v1-contract/contracts/libraries/HedgepieLibrary.sol [hunk 3]
```diff
@@ -1,10 +1,8 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 pragma solidity ^0.8.4;
 
-import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
-import "@uniswap/v3-periphery/contracts/interfaces/INonfungiblePositionManager.sol";
-import "../libraries/SafeBEP20.sol";
-import "../libraries/Ownable.sol";
+import "./SafeBEP20.sol";
+import "./Ownable.sol";
```
Comment (daiki-iu) [inline v1-contract/contracts/libraries/HedgepieLibrary.sol:6]: @naruto0913  we don't need any change in this file.

File: v1-contract/contracts/interfaces/IAdapterManager.sol [hunk 4]
```diff
@@ -60,4 +60,22 @@ interface IAdapterManager {
...
+        );
+
+    function getRemoveLiqCallData(address _adapter, uint256 _amount)
+        external
+        view
+        returns (
+            address to,
+            uint256 value,
+            bytes memory data
+        );
```
Comment (daiki-iu) [inline v1-contract/contracts/interfaces/IAdapterManager.sol:80]: @naruto0913  can we add IAdapterManagerMatic file?

**Review by daiki-iu — CHANGES_REQUESTED** [review]

File: v1-contract/contracts/interfaces/IAdapterManager.sol (hunk 4 above)
Comment (naruto0913) [reply v1-contract/contracts/interfaces/IAdapterManager.sol:80]: Ok

File: v1-contract/contracts/libraries/HedgepieLibrary.sol (hunk 3 above)
Comment (naruto0913) [reply v1-contract/contracts/libraries/HedgepieLibrary.sol:6]: Ok, got you

File: v1-contract/contracts/HedgepieInvestor.sol (hunk 2 above)
Comment (naruto0913) [reply v1-contract/contracts/HedgepieInvestor.sol:8]: Ok

**Review by daiki-iu — APPROVED** [review]
@naruto0913  LGTM!

**Pull request merged**